pkg/nodes/node.go

```diff
@@ -189,6 +219,32 @@ func (builder *NodeBuilder) ExternalIPv4Network() (string, error) {
 	return extNetwork.IPv4, nil
 }
 
+// WaitUntilNodeStable waits for the duration of the defined timeout or until the
+// node gets to a stable - not unschedulable.
+func (builder *NodeBuilder) WaitUntilNodeStable(timeout time.Duration) error {
+	glog.V(100).Infof("Waiting for the defined period until node %s is not unschedulable",
+		builder.Definition.Name)
+
+	if builder.errorMsg != "" {
+		return fmt.Errorf(builder.errorMsg)
+	}
+
+	// Polls every retryInterval to determine if the node is not unschedulable.
+	return wait.PollImmediate(time.Second, timeout, func() (bool, error) {
+		node, err := builder.Get()
+
+		if err != nil {
+			return false, nil
+		}
+
+		if !node.Spec.Unschedulable {
```

Comment:
I think you need to make sure that node is in Schedulable state for some specific period of time, sometimes we mark node as schedulable and them after 5 seconds move to unscheduled

Reply:
Please refer to: https://github.com/openshift-kni/eco-goinfra/blob/b2486f52938008d0fa468e4e38a852ec60e9027f/pkg/mco/mcp.go#L248

Reply:
No you need to add 1 more PollImmediate function with additional stable timeout. Please check the reference link that I've share with you